crates/pepper/src/catalog.rs

```diff
@@ -0,0 +1,174 @@
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//! Trait abstraction for metadata catalog operations.
+//!
+//! This trait defines the interface for managing table metadata
+//! and file references. It can be implemented by different RDBMS backends
+//! (`SQLite`, `PostgreSQL`, etc.).
+
+use super::metadata::{CreateTableOptions, DataFile, DeleteFile, TableMetadata, TableStats};
+use async_trait::async_trait;
+use snafu::Snafu;
+use std::sync::Arc;
+
+/// Error type for catalog operations.
+#[derive(Debug, Snafu)]
+pub enum CatalogError {
+    /// Database error
+    #[snafu(display("Database error: {message}"))]
+    Database {
+        /// Error message
+        message: String,
+    },
+
+    /// Table not found
+    #[snafu(display("Table not found: {table_name}"))]
+    TableNotFound {
+        /// Name of the table that was not found
+        table_name: String,
+    },
+
+    /// Table already exists
+    #[snafu(display("Table already exists: {table_name}"))]
+    TableAlreadyExists {
+        /// Name of the table that already exists
+        table_name: String,
+    },
+
+    /// Invalid operation
+    #[snafu(display("Invalid operation: {message}"))]
+    InvalidOperation {
+        /// Description of the invalid operation
+        message: String,
+    },
+
+    /// IO error
+    #[snafu(display("IO error: {source}"))]
+    Io {
+        /// The underlying IO error
+        source: std::io::Error,
+    },
+
+    /// `SQLite` error
+    #[snafu(display("SQLite error: {source}"))]
+    Sqlite {
+        /// The underlying `SQLite` error
+        source: rusqlite::Error,
+    },
+
+    /// Task join error
+    #[snafu(display("Task join error: {source}"))]
+    TaskJoin {
+        /// The underlying task join error
+        source: tokio::task::JoinError,
+    },
+}
+
+// Implement From trait for rusqlite::Error
+impl From<rusqlite::Error> for CatalogError {
+    fn from(err: rusqlite::Error) -> Self {
+        CatalogError::Sqlite { source: err }
+    }
+}
+
+// Implement From trait for tokio::task::JoinError
+impl From<tokio::task::JoinError> for CatalogError {
+    fn from(err: tokio::task::JoinError) -> Self {
+        CatalogError::TaskJoin { source: err }
+    }
+}
+
+// Implement From trait for std::io::Error
+impl From<std::io::Error> for CatalogError {
+    fn from(err: std::io::Error) -> Self {
+        CatalogError::Io { source: err }
+    }
+}
+
+/// Result type for catalog operations.
+pub type CatalogResult<T> = Result<T, CatalogError>;
```

Comment:
```suggestion
pub type Result<T, E = CatalogError> = Result<T, E>;
```

Unless there are multiple error types in a module, we define this to be `Result`: https://github.com/spiceai/spiceai/blob/trunk/docs/dev/style_guide.md#establishing-the-result-type-alias-in-each-module